src/backend/trigger_builder.rs

```diff
@@ -0,0 +1,61 @@
+use crate::*;
+
+pub trait TriggerBuilder: TableRefBuilder + QueryBuilder {
+    /// Translate [`TriggerCreateStatement`] into SQL statement.
+    fn prepare_trigger_create_statement(
+        &self,
+        create: &TriggerCreateStatement,
+        sql: &mut dyn SqlWriter,
+    ) {
+        write!(sql, "CREATE TRIGGER ").unwrap();
+        self.prepare_create_trigger_if_not_exists(create, sql);
+
+        let trigger_ref = match &create.trigger.name {
+            Some(value) => value,
+            // auto-generate trigger name
+            _ => &create.trigger.trigger_ref(),
+        };
+        let trigger_ref: TableRef = trigger_ref.into();
+        self.prepare_table_ref_iden(&trigger_ref, sql);
+        write!(sql, " {} {} ON ", create.trigger.time, create.trigger.event).unwrap();
+        self.prepare_table_ref_iden(&create.trigger.table, sql);
+        write!(sql, " FOR EACH ROW\nBEGIN\n").unwrap();
```

Comment:
It would be nice to have statement level triggers although they aren't available in mysql.